### PR TITLE
tests: introduce Fedora ELN test scenario

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Review by VladimirSlavik — COMMENTED** [review]
What's the plan with the koji builds and bodhi updates? I mean, anaconda cannot update after GA, so what will this do?

**Comment by KKoukiou** [comment]
> What's the plan with the koji builds and bodhi updates? I mean, anaconda cannot update after GA, so what will this do?

Oh this first commit is already merged through another PR.  Wil rebase and will disappear. This acts up on top of dist-git commits from packit. If we don't create releases, thus no dist-git PRs, this will do nothing.

**Review by VladimirSlavik — APPROVED** [review]

**Comment by martinpitt** [comment]
I am trying to reproduce the -boot image failures locally. Unfortunately that image doesn't set up a serial console, so `virsh console` doesn't work, and thus we cannot capture/log boot failures. I'll see what's going on with virt-viewer, once it actually fails.

I am trying with
```
TEST_OS=fedora-rawhide-boot make integration-test
TEST_AUDIT_NO_SELINUX=1 TEST_OS=fedora-rawhide-boot test/common/run-tests TestBasic.testAboutModal
```
which should be pretty much what CI does..

However, I do notice that even on my quiet laptop "start booting" to "SSH responds" takes 1:10 to 1:30 minutes. That is darn close to the default timeout of 2 minutes, especially on loaded CI machines. For example, even the initial part between "start qemu" and "boot loader downloads updates.img" takes 15s on my machine, and 43s on CI [here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-70-20231205-123935-031f44f8-fedora-rawhide-boot-bots-5585/log).

I pushed an additional commit which bumps the timeout, let's see if that helps.

**Comment by martinpitt** [comment]
So that apparently did help for rawhide-boot -- the [test shows a flake](https://cockpit-logs.us-east-1.linodeobjects.com/pull-70-20231205-171642-23e731db-fedora-rawhide-boot-bots-5585/log.html), but unrelated to the boot failure. Retrying to get some more data for boot stability.

But didn't help for [ELN boot](https://cockpit-logs.us-east-1.linodeobjects.com/pull-70-20231205-171649-23e731db-fedora-eln-boot-bots-5585/log), I'll see if that reproduces locally.

File: test/machine_install.py
Comment (martinpitt) [inline test/machine_install.py:172]: It's necessary because of the one-minute grub menu timeout -- so that's just sitting around idly. Update the commit message for that.

**Comment by martinpitt** [comment]
As a first check, I validated that `./vm-run fedora-eln-boot` works. It needs a little fix: https://github.com/cockpit-project/bots/pull/5641 but there's no root fs corruption.

It's not how we invoke qemu etc. in our CI, it is related to updates.img. When disabling that:
```
--- test/machine_install.py
+++ test/machine_install.py
@@ -153,7 +153,8 @@ class VirtInstallMachine(VirtMachine):
                 "--noautoconsole "
                 f"--graphics vnc,listen={self.ssh_address} "
                 "--extra-args "
-                f"'inst.sshd inst.webui.remote inst.webui inst.updates=http://10.0.2.2:{self.http_updates_img_port}/updates.img' "
+                # f"'inst.sshd inst.webui.remote inst.webui inst.updates=http://10.0.2.2:{self.http_updates_img_port}/updates.img' "
+                f"'inst.sshd inst.webui.remote inst.webui' "
                 "--network none "
                 f"--qemu-commandline="
                 "'-netdev user,id=hostnet0,"
```

then fedora-eln-boot starts up fine. Of course it doesn't start anaconda-webui but the classic GTK installer.

I tried to bump the memory from 4096 MB to 6000 MB, and that just causes virt-viewer to disconnect, and when reconnecting I see "no bootable device". So it's not just a question of RAM, that ginormous updates.img somehow trips up dracut/overlayfs/etc.

Throwing out some unnecessary junk from updates.img seems to help, though -- I see the image booting into webui and some initial tests passing. I added an extra commit for that.

**Comment by martinpitt** [comment]
@KKoukiou @jkonecny12 BTW, please enable branch protection rules. This PR stays at "approved" even after several force pushes and new commits.

File: test/machine_install.py
Comment (martinpitt) [reply test/machine_install.py:172]: done.

**Comment by martinpitt** [comment]
@KKoukiou One more thing that needs fixing: the [f-e-boot run](https://cockpit-logs.us-east-1.linodeobjects.com/pull-70-20231206-093154-ecb08eff-fedora-eln-boot/log) downloads and uses the wrong payload:

> bots/image-download fedora-rawhide-anaconda-payload
> 127.0.0.1 - - [06/Dec/2023 09:43:36] "HEAD /fedora-rawhide-anaconda-payload-b82e7eddd32b4faa15ddb5701ab3f3ebcbb05ea64f2108f1cb0f9d1e88d5b213.tar.gz HTTP/1.1" 200 -

Doesn't have to block this PR, but a follow-up would be good?

**Pull request merged**
